src/main/java/com/starrocks/connector/flink/manager/StarRocksSinkManager.java

```diff
@@ -425,17 +433,54 @@ private void validateTableStructure(TableSchema flinkSchema) {
         if (sinkOptions.hasColumnMappingProperty()) {
```

Comment:
if not set sink.properties.columns，partial update should not be work?

Reply:
Thx for the warning. so i should change be 
~~~java

        if (this.sinkOptions.getSinkPartialUpdate()) {
            this.checkPartialColumnMode(flinkSchema, rows);
        } else {
            if (this.sinkOptions.hasColumnMappingProperty()) {
                return;
            }
            this.checkAllColumnMode(flinkSchema, rows);
        }

~~~
right ?

Reply:
i means that if set partial update but not set columns，sr should not accept

Reply:
got it

Reply:
done

Reply:
I think @xlfjcg means that we need add "column" to the HTTP header automatically for partial update. Please refer to [StarRocks Partial Update ](https://docs.starrocks.io/zh-cn/latest/loading/PrimaryKeyLoad#%E9%83%A8%E5%88%86%E6%9B%B4%E6%96%B0%E5%85%AC%E6%B5%8B%E4%B8%AD) for details. You can refer to [StarRocksStreamLoadVisitor#doHttpPut](https://github.com/StarRocks/starrocks-connector-for-apache-flink/blob/main/src/main/java/com/starrocks/connector/flink/manager/StarRocksStreamLoadVisitor.java#L240) to see how to add "column" to the Http header.

Reply:
got it, updated